x-pack/plugins/maps/public/store/map.js

```diff
@@ -94,8 +101,19 @@ const INITIAL_STATE = {
   waitingForMapReadyLayerList: [],
 };
 
+
+
 export function map(state = INITIAL_STATE, action) {
   switch (action.type) {
+    case REMOVE_TRACKED_LAYER_STATE:
+      return removeTrackedLayerState(state, action.layerId);
+    case ROLLBACK_ANY_TRAILING_TRACKER_LAYER_STATE:
+      rollbackAnyTrailingTrackedLayerState(state);//do this because users can select different layers while the layer panel is open, which automatically switches the selection
```

Comment:
This will need to return the results of `rollbackAnyTrailingTrackedLayerState`. Only reason its still working is that `rollbackAnyTrailingTrackedLayerState` is still modifying state.